text/0107-construct-library-module-lifecycle.md

```diff
@@ -0,0 +1,146 @@
+---
+feature name: aws-construct-library-module-lifecycle
+start date: 2020-01-29
+rfc pr: (leave this empty)
+related issue: 0107
+---
+
+# Summary
+
+The purpose of this document is to outline the process used to prioritize and mature modules in the AWS-CDK Construct Library. It outlines how we graduate modules from CloudFormation-only resources (L1) to modules that contain high-level resource abstractions (L2) and stabilize them for GA release.
+
+# Motivation
+
+We want to be transparent about the processes the AWS CDK team use, and seek community input to make them better.
+
+# Stages of an AWS-CDK Construct Library Module
+
+The sections below outline the development stages for modules in the AWS-CDK Construct Library and the criteria to graduate the module to the next level of maturity. There are 4 stages of development for an AWS-CDK Construct Library module: 
+
+* Stage 0 - CFN resources (L1) only 
+* Stage 1 - Experimental
+* Stage 2 - Developer Preview
+* Stage 3 - GA 
+
+## Stage 0 - CFN resources (L1) only
+
+All construct library modules start in stage 0 when they are auto-generated from the CloudFormation resource specification. The goal of stage 0 is to make new CloudFormation resources/properties available to CDK customers as soon as possible, and create tracking artifacts that allow us to capture the data required to make prioritization decisions to add L2 resources in the future. 
+
+* **Stage 0 activities:**
+    * **Publish package**  - a package for the module is autogenerated and published in the aws-cdk GitHub repo
+    * **Publish README** - the module has a rudimentary README doc that: 
+        * enables the [Stability:Stable] tag
+        * links to the relevant CloudFormation docs
+        * includes the basic import statement 
+    * **Update API reference** - list the module in the AWS Construct Library API reference
```

Comment:
What do you mean? Doesn't this happen automatically?

Reply:
It does happen automatically. This item is on the list for completeness, since the API reference is important. Do we have (or need) a test that verifies the API reference is created for new construct modules? 

For example - I don't want this to happen anymore, [fixture README](https://docs.aws.amazon.com/cdk/api/latest/docs/fixture-readme.html)

Reply:
Right, so this stage should be 'verify that the docs are rendered correctly' with a good definition of what we mean by 'correctly'.

However, the bit I'm missing is - who is going to check and how are they going to know that something needs to be checked? Looks like we can build some automation around this.